src/jit/emitarm64.cpp

```diff
@@ -5347,6 +5348,22 @@ void emitter::emitIns_R_R_R_I(instruction ins,
     id->idReg2(reg2);
     id->idReg3(reg3);
 
+    if (attrReg2 != EA_UNKNOWN)
+    {
+        if (EA_IS_GCREF(attrReg2))
```

Comment:
You could add this assert here:

assert((fmt == IF_LS_3B) || (fmt == IF_LS_3C));